test-suite/src/function/format.rs

```diff
@@ -0,0 +1,31 @@
+use crate::*;
+
+test_case!(format, async move {
+    use gluesql_core::{executor::EvaluateError, prelude::Value::*};
+
+    let test_cases = vec![
+        (
+            r#"SELECT FORMAT(DATE "2017-06-15", "%Y") AS DATE"#,
+            Ok(select!(
+                "DATE"
+                Str;
+                "2017".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT FORMAT(TIMESTAMP "2015-09-05 23:56:04", "%Y-%m-%d %H") AS TIMESTAMP"#,
+            Ok(select!(
+                "TIMESTAMP"
+                Str;
+                "2015-09-05 23".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT FORMAT("2015-09-05 23:56:04", "%Y-%m-%d %H") AS TIMESTAMP"#,
+            Err(EvaluateError::FunctionRequiresFormattableValue("FORMAT".to_string()).into()),
+        ),
```

Comment:
It would be nice if there are plenty of test cases.
eg1) If I don't know how to format minute and second, i could glance test cases
eg2) Someone may want to get `Day of week`
![image](https://user-images.githubusercontent.com/51254761/188473368-c7e6abc1-2ee9-4ef1-b1bb-bb2e6f9728f9.png)
(https://livesql.oracle.com/apex/livesql/file/content_GCEY1DN2CN5HZCUQFHVUYQD3G.html)

Reply:
As we are using `Chrono` to format the date I read through the document of the `Chrono` and found [this](https://docs.rs/chrono/latest/chrono/format/strftime/index.html). I will try to add the cases with the use of the following document.  But I am still not sure whether I should put all of the possible cases to our unit test.

Reply:
We don't have to consider every case of `strftime`. (This is because `chrono` is doing well)
Instead, I think we should decide how to handle an invalid argument in the format argument.
What does the `format` method of NativeDate and DateTime currently used do when an invalid argument is received?

Reply:
I agree it can be redundant to list all cases but i thought each representative of usages can have meaning.
Or at least [format table](https://docs.rs/chrono/latest/chrono/format/strftime/index.html) you found should be included in our doc someday.

Reply:
> We don't have to consider every case of `strftime`. (This is because `chrono` is doing well) Instead, I think we should decide how to handle an invalid argument in the format argument. What does the `format` method of NativeDate and DateTime currently used do when an invalid argument is received?

It currently returns the error message that `Chrono` returns.

Reply:
when the invalid argument is given, the `Format` returns the wrong argument.
for example if `Y-M` is given to the second argument  which is invalid, it returns 
`Ok(Select { labels: ["TIMESTAMP"], rows: [[Str("Y-M")]] })`